### DISCUSSION TITLE
More than one base link?

### DISCUSSION THREAD

**Discussion by DaniCoimbra** [NVlabs/curobo#296] [Software Q&A]
Hello, in my robot configuration I need two links to be connected to world and fixed, how can I setup that robot?
A fixed joint with the two links creates a closed articulation, and 2 joints connected to "World" also creates a closed articulation...

Is there a known way to implement this?

edit: in other words, is curobo limited to open articulations? are closed articulations supported?

**Comment by balakumar-s** [comment]
Closed kinematic chains are not supported. However, some closed chains can be formulated as open chains with mimic joints. One example is the robotiq 2f-140 gripper. Here is how it's represented with mimic joints: https://github.com/NVlabs/curobo/blob/0c51dd2da8d4a76ebaa0dee4e302ef88e504bce0/src/curobo/content/assets/robot/ur_description/ur5e_robotiq_2f_140.urdf#L624
If your closed chain can be represented as an open chain with mimic joints, then you could use cuRobo. 


From your question, it looks like you have two fixed joints connecting 1 link (hip-link?) to the ground plane through 2 links (legs?). Does it make sense to use hip-link as the base_link and convert everything to be w.r.t. hip-link? Then there is no closed kinematic chain.
